.github/workflows/generate_new_client.yaml

```diff
@@ -0,0 +1,140 @@
+        id: date
+        run: echo "::set-output name=week_of_year::$(date +'%W' --utc)"
+      - name: Install new-client.py dependencies
+        run: pip install --require-hashes -r generation/new_client/requirements.txt
+      - run: echo 'a message'
+      - name: Generate
+        id: generation
+        run: |
+          set -x
+          arguments="--api_shortname=\"${API_SHORTNAME}\" \
+          --proto-path=\"${PROTO_PATH}\" \
+          --name-pretty=\"${NAME_PRETTY}\" \
+          --product-docs=\"${PRODUCT_DOCS}\" \
+          --api-description=\"${API_DESCRIPTION}\""
+
+          # helper function that appends a python argument only if specified in the GH action inputs
+          append_argument() {
+            py_arg=$1
+            # env vars look exactly like new-client arguments but uppercase + underscores
+            env_name=$(echo "${py_arg}" | sed 's/-/_/g' | sed -e 's/\([a-z]\)/\U\1/g')
+            if [[ -n "${!env_name}" ]]; then
+              # $(echo) is redundant but it works around a syntax highlighting problem in vim
+              arguments=$(echo "${arguments} --${py_arg}=\"${!env_name}\"")
+            fi
+          }
+
+          declare -a optional_args=('transport' 'cloud-api' 'group-id' 'rest-docs' 'rpc-docs')
+
+          for python_argument in "${optional_args[@]}"; do
+            append_argument "${python_argument}"
+          done
+          echo "::set-output name=new_library_args::${arguments}"
+          echo "${arguments} --googleapis-gen-url=\"${GOOGLEAPIS_GEN_URL}\"" | xargs python generation/new_client/new-client.py generate
+        env:
+          GOOGLEAPIS_GEN_URL: https://yoshi-approver:${{ secrets.YOSHI_CODE_BOT_TOKEN }}@github.com/googleapis/googleapis-gen.git
+          API_SHORTNAME: ${{ github.event.inputs.api_shortname }}
+          NAME_PRETTY: ${{ github.event.inputs.name_pretty }}
+          PROTO_PATH: ${{ github.event.inputs.proto_path }}
+          PRODUCT_DOCS: ${{ github.event.inputs.product_docs }}
+          REST_DOCS: ${{ github.event.inputs.rest_docs }}
+          RPC_DOCS: ${{ github.event.inputs.rpc_docs }}
+          API_DESCRIPTION: ${{ github.event.inputs.api_description }}
+          TRANSPORT: ${{ github.event.inputs.transport }}
+          CLOUD_API: ${{ github.event.inputs.CLOUD_API }}
+          GROUP_ID: ${{ github.event.inputs.group_id }}
+      - name: Push to branch and create PR
+        run: |
+          set -x
+          [ -z "`git config user.email`" ] && git config --global user.email "${USERNAME:-script}@google.com"
+          [ -z "`git config user.name`" ] && git config --global user.name "${USERNAME:-script}"
+
+          # create and push to branch in origin
+          branch_name="new-library/${{ github.event.inputs.api_shortname }}"
+          git checkout -b "${branch_name}"
+          git add --all
+          git commit -m "feat: [${API_SHORTNAME}] new module for ${API_SHORTNAME}
+
+
+          \`\`\`${GENERATION_ARGUMENTS}\`\`\`"

```

Comment:
(Just in case you're doing this) Avoid experimenting the git command in this pull request because it's time-consuming. You have to wait minutes to see the results. Try these git and gh commands locally before editing this pull request.